python/pyspark/sql/tests/test_interchange.py

```diff
@@ -0,0 +1,54 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import unittest
+import pyarrow as pa
+import pandas as pd
+import pyspark.pandas as ps
+
+try:
+    import duckdb
+
+    DUCKDB_TESTS = True
+except ImportError:
+    DUCKDB_TESTS = False
+
+
+@unittest.skipIf(not DUCKDB_TESTS, "duckdb is not installed")
```

Comment:
Is the only way to test this with `duckdb`? We are introducing a new library just to test a feature. If it implements an arrow protocol, shouldn't we just be able to test with arrow or something?

Reply:
I added a test for arrow specifically, but I think it's still worth checking `duckdb`. arrow -> arrow is good to test as well, but there is power in testing integration. We can add more over time, but duckdb is a solid one to start with.